packages/php-wasm/node/src/lib/load-runtime.ts

```diff
@@ -111,32 +123,46 @@ export async function loadNodeRuntime(
 								phpRuntime.FS.filesystems.NODEFS.realPath(node)
 							)
 						);
-					const symlinkPath = joinPaths(
+					const symlinkMountPath = joinPaths(
 						`/internal/symlinks`,
 						absoluteSourcePath
 					);
-					if (
-						!FSHelpers.fileExists(phpRuntime.FS, symlinkPath) &&
-						fs.existsSync(absoluteSourcePath)
-					) {
-						const sourceStat = fs.statSync(absoluteSourcePath);
-						if (sourceStat.isDirectory()) {
-							phpRuntime.FS.mkdirTree(symlinkPath);
-						} else if (sourceStat.isFile()) {
-							phpRuntime.FS.mkdirTree(dirname(symlinkPath));
-							phpRuntime.FS.writeFile(symlinkPath, '');
-						} else {
-							throw new Error(
-								'Unsupported file type. PHP-wasm supports only symlinks that link to files, directories, or symlinks.'
+					if (fs.existsSync(absoluteSourcePath)) {
+						if (
+							!FSHelpers.fileExists(
+								phpRuntime.FS,
+								symlinkMountPath
+							)
+						) {
+							const sourceStat = fs.statSync(absoluteSourcePath);
+							if (sourceStat.isDirectory()) {
+								phpRuntime.FS.mkdirTree(symlinkMountPath);
+							} else if (sourceStat.isFile()) {
+								phpRuntime.FS.mkdirTree(
+									dirname(symlinkMountPath)
+								);
+								phpRuntime.FS.writeFile(symlinkMountPath, '');
+							} else {
+								throw new Error(
+									'Unsupported file type. PHP-wasm supports only symlinks that link to files, directories, or symlinks.'
+								);
+							}
+						}
+
+						const symlinkMountNode =
+							phpRuntime.FS.lookupPath(symlinkMountPath).node;
+						if (
+							symlinkMountNode.mount.mountpoint !==
```

Comment:
I'm a bit lost in this check. Would you mind adding some commentary? Is it useful because `lookupPath()` follows symlinks and will return the final path in the chain?